### ISSUE TITLE
Grafana Crashes with new version of Clickhouse

### ISSUE THREAD

**Issue by misterek** [grafana/clickhouse-datasource#683]
**What happened**:
Starting around 12 hours ago, new grafana pods crash.  

The last logs output are:
```
logger=grafana.update.checker t=2024-02-06T12:50:42.853890535Z level=info msg="Update check succeeded" duration=30.237941ms
logger=plugins.update.checker t=2024-02-06T12:50:42.886804646Z level=info msg="Update check succeeded" duration=63.142425ms
```
I can't give any strong evidence, but by process of elimination, removing the clickhouse plugin allowed grafana to start. This was the only change that was made to allow it to start.

**What you expected to happen**:

A plugin update to not cause grafana to crash.

**How to reproduce it (as minimally and precisely as possible)**:

Apologies, but I'm going to be unlikely to articulate this.  I notice that there were a couple of releases after 4.0.0 to address things, perhaps it's related. What I can say is that pods that had started 36h ago (which still looks to be after the latest github release) did _not_ crash, but pods starting in the last 12 hour or so _did_ crash.

I probably won't be able to do much/any additional troubleshooting here, but did want to open a bug report.

**Screenshots**

**Anything else we need to know?**:

**Environment**:

- Grafana version: 10.3.1
- Plugin version: Presumably the latest
- OS Grafana is installed on: grafana docker image
- User OS & Browser:N/A
- Others:

**Comment by mshustov** [comment]
I tested
- the build from `main` branch against `main` branch of Grafana
- the build from `main` branch against `latest` Grafana image
- plugin `v4.0.2` with `latest` Grafana image
- plugin `v4.0.2` on Grafana Cloud

and didn't manage to spot any problems.

@misterek do you have any logs? Could you provide your configuration for the plugin?

**Comment by jkaflik** [comment]
@misterek could you provide what is process exit code?

**Comment by asimpson** [comment]
I attempted to reproduce this as well. I ran: `docker run -v path/to/plugins:/var/lib/grafana/plugins/ -v path/to/data:/var/lib/grafana -p 3001:3000 grafana/grafana:10.3.1`

I installed the current Clickhouse release, 4.0.2, and configured a data source.

Quitting the Docker process and spinning it back up works fine.

---

@misterek you mention it was presumably the latest version of Clickhouse, can you verify that? If it's not 4.0.2, what version is it?

**Comment by misterek** [comment]
I'm sorry.  I totally forgot to respond. 

Unfortunately, I don't think I'll be able to go back and reproduce, but here's the additional detail I can think of:

* Our usage of this plugin was _very_ light.  We only had one clickhouse data source, and it was only used in a single dashboard.
* The times noted above -- we saw the pods restart ~ 36 hours prior to the problem.  This was _after_ the latest clickhouse release.  They started just fine.
* Shortly around the time noted above, the pods started crashing, with the last log lines being as above.  No useful error information.
* Process of elimination (removing plugins 1 by 1), we ended up removing the clickhouse plugin.  That was the only change required fo the pods to continue starting.

It certainly could have been some odd mix of plugins that caused weirdness, but in the end, the only change between crashing and starting was removing the clickhouse plugin.

Apologies that I can't be of much more help. I did want to report it in case others were having it, but I'll go ahead and close this since I probably can't give you enough information to be of help.  Thanks!
